### PR TITLE
Remove dependency on RSQLite

### PR DESCRIPTION
**Pull request by ablack3**
Remove dependency on RSQLite and use Andromeda interface for queryToAndromeda functions.

There is one change that should require the current develop branch of Andromeda (see comment below).

`names(andromeda[[andromedaTableName]]) <- newColumnNames`

Unfortunately I don't think these functions are covered by tests. @schuemie, If you're ok with these changes conceptually then I'll write some tests as a next step.

### CODE REVIEW

File: R/Andromeda.R [hunk 1]
```diff
@@ -102,13 +102,11 @@ lowLevelQuerySqlToAndromeda.default <- function(connection,
       integerAsNumeric = integerAsNumeric
     )
```
Comment (ablack3) [inline R/Andromeda.R:104]: Here I just use andromeda's interface for writing and appending to a table.

File: R/Andromeda.R [hunk 2]
```diff
@@ -239,21 +231,12 @@ querySqlToAndromeda <- function(connection,
         integerAsNumeric = integerAsNumeric,
         integer64AsNumeric = integer64AsNumeric
       )
-      columnNames <- RSQLite::dbListFields(andromeda, andromedaTableName)
+      columnNames <- colnames(andromeda[[andromedaTableName]])
```
Comment (ablack3) [inline R/Andromeda.R:234]: This should work with the current Andromeda version.

File: R/Andromeda.R [hunk 3]
```diff
@@ -239,21 +231,12 @@ querySqlToAndromeda <- function(connection,
...
-      if (any(idx)) {
-        sql <- sprintf(
-          "ALTER TABLE %s RENAME COLUMN %s TO %s;",
-          andromedaTableName,
-          columnNames[idx],
-          newColumnNames[idx]
-        )
-        lapply(sql, function(x) RSQLite::dbExecute(andromeda, x))
-      }
+      names(andromeda[[andromedaTableName]]) <- newColumnNames
```
Comment (ablack3) [inline R/Andromeda.R:239]: This required a new Andromeda method and is in the current develop branch of Andromeda.

**Comment by ablack3** [comment]
The R-CMD checks are failing because of this error   
cannot open URL 'https://api.github.com/repos/ohdsi/SqlRender/contents/DESCRIPTION?ref=develop'

**Comment by ablack3** [comment]
> @ablack3 Am I correct in thinking that this branch will mean that any db backend can be used with andromeda, including remote backends?

@azimov Basically I need to make sure that if other packages use Andromeda they do so through it's interface (functions that are tested) and not rely on implementation details like the fact that it uses SQLite under the hood. I'm working on a new version of Andromeda that uses duckdb instead of Sqlite. 

I had been thinking of Andromeda as a local database but Martijn's view of it is really as a replacement for ff which is strictly to support on-disk dataframes and not really treat it as a database. I'd love to hear your use cases or thoughts about Andromeda. Have you seen the dm package (https://cynkra.github.io/dm/)?

**Comment by azimov** [comment]
> I'd love to hear your use cases or thoughts about Andromeda. Have you seen the dm package (https://cynkra.github.io/dm/)?

I've not seen DM but that looks incredibly useful, thanks!

>  I had been thinking of Andromeda as a local database but Martijn's view of it is really as a replacement for ff which is strictly to support on-disk dataframes and not really treat it as a database.

My use case is simply using FeatureExtraction with a huge number of cohorts where storing results on a disk isn't desirable or efficient, this is sort of supported in FeatureExtraction for a limited subset of functions (and it seems buggy and inconsistent so I just wrote this [PR](https://github.com/OHDSI/FeatureExtraction/pull/153)). If Andromeda supported using any Sql db then my thought was that no minimal would need to be made to FE to support this mode of working.

I fully accept that this isn't want Andromeda was designed to do and I'm definitely not proposing it, this commit just made me think it would be possible.

**Comment by ablack3** [comment]
> If Andromeda supported using any Sql db then my thought was that no minimal would need to be made to FE to support this mode of working.

I think the goal of Andromeda is to support larger than memory dataframes/tables (including multiple related tables) in R and writing FeatureExtraction results directly to a remote database should be handled in FeatureExtraction.

**Pull request merged**